x/pubkey/client/cli/tx.go

```diff
@@ -0,0 +1,202 @@
+		Use:                        types.ModuleName,
+		Short:                      fmt.Sprintf("%s transactions subcommands", types.ModuleName),
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+	cmd.AddCommand(
+		AddKey(valAddrCodec),
+	)
+	return cmd
+}
+
+// AddKey returns the command for generating the SEDA keys and
+// uploading their public keys on chain.
+func AddKey(ac address.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "add-seda-keys",
+		Short: "Generate the SEDA keys and upload their public keys.",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			serverCfg := server.GetServerContextFromCmd(cmd).Config
+
+			fromAddr := clientCtx.GetFromAddress()
+			if fromAddr.Empty() {
+				return fmt.Errorf("set the from address using --from flag")
+			}
+			valAddr, err := ac.BytesToString(fromAddr)
+			if err != nil {
+				return err
+			}
+
+			var pks []types.IndexedPubKey
+			keyFile, _ := cmd.Flags().GetString(FlagKeyFile)
+			if keyFile != "" {
+				pks, err = loadSEDAPubKeys(keyFile)
+				if err != nil {
+					return err
+				}
+			} else {
+				pks, err = generateSEDAKeys(
+					[]privKeyGenerator{secp256k1GenPrivKey},
+					filepath.Dir(serverCfg.PrivValidatorKeyFile()),
+				)
+				if err != nil {
+					return err
+				}
+			}
+
+			msg := &types.MsgAddKey{
+				ValidatorAddr:  valAddr,
+				IndexedPubKeys: pks,
+			}
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	cmd.Flags().String(FlagKeyFile, "", "path to an existing SEDA key file")
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+type IndexedPrivKey struct {
+	Index   uint32            `json:"index"`
+	PrivKey cmtcrypto.PrivKey `json:"priv_key"`
+}
+
+// loadSEDAPubKeys loads the SEDA key file from the given path and
+// returns a list of index-public key pairs.
+func loadSEDAPubKeys(loadPath string) ([]types.IndexedPubKey, error) {
+	keysJSONBytes, err := os.ReadFile(loadPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read SEDA keys from %v: %v", loadPath, err)
+	}
+	var keys []IndexedPrivKey
+	err = cmtjson.Unmarshal(keysJSONBytes, &keys)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal SEDA keys from %v: %v", loadPath, err)
+	}
+
+	result := make([]types.IndexedPubKey, len(keys))
+	for i, key := range keys {
+		// Convert to SDK type for app-level use.
+		pubKey, err := cryptocodec.FromCmtPubKeyInterface(key.PrivKey.PubKey())
+		if err != nil {
+			return nil, err
+		}
+		pkAny, err := codectypes.NewAnyWithValue(pubKey)
+		if err != nil {
+			return nil, err
+		}
+		result[i] = types.IndexedPubKey{
+			Index:  key.Index,
+			PubKey: pkAny,
+		}
+	}
+	return result, nil
+}
+
+// saveSEDAKeys saves a given list of IndexedPrivKey in the directory
+// at dirPath.
+func saveSEDAKeys(keys []IndexedPrivKey, dirPath string) error {
```

Comment:
These are private keys right? This is fine for devnet. Cosmos offers the file backend for a similar case, but they do specify you should NOT use file storage for an official deployment.

This security stuff could also done in another pr. Also maybe we could leverage the existing system cosmos has to use a different storage backed.

Reply:
created an issue #368